src/helpers/upgrade/executableHelpers.ts

```diff
@@ -0,0 +1,135 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { NativefierOptions } from '../../options/model';
+import { getVersionString } from './rceditGet';
+type ExecutableInfo = {
+  arch?: string;
+};
+
+function getExecutableBytes(executablePath: string): Uint8Array {
+  return fs.readFileSync(executablePath);
+}
+
+function getExecutableArch(
+  exeBytes: Uint8Array,
+  platform: string,
+): string | undefined {
+  switch (platform) {
+    case 'linux':
+      // https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+      switch (exeBytes[0x12]) {
+        case 0x03:
+          return 'ia32';
+        case 0x28:
+          return 'armv7l';
+        case 0x3e:
+          return 'x64';
+        case 0xb7:
+          return 'arm64';
+        default:
+          return undefined;
+      }
+    case 'darwin':
+    case 'mas':
+      // https://opensource.apple.com/source/xnu/xnu-2050.18.24/EXTERNAL_HEADERS/mach-o/loader.h
+
+      switch ((exeBytes[0x04] << 8) + exeBytes[0x05]) {
+        case 0x0700:
+          return 'x64';
+        case 0x0c00:
+          return 'arm64';
+        default:
+          return undefined;
+      }
+    case 'windows':
+      // https://en.wikibooks.org/wiki/X86_Disassembly/Windows_Executable_Files#COFF_Header
+      switch ((exeBytes[0x7d] << 8) + exeBytes[0x7c]) {
+        case 0x014c:
+          return 'ia32';
+        case 0x8664:
+          return 'x64';
+        case 0xaa64:
+          return 'arm64';
+        default:
+          return undefined;
+      }
+    default:
+      return undefined;
+  }
+}
+
+function getExecutableInfo(
+  executablePath: string,
+  platform: string,
+): ExecutableInfo {
+  const exeBytes = getExecutableBytes(executablePath);
+  return {
+    arch: getExecutableArch(exeBytes, platform),
+  };
+}
+
+export function getOptionsFromExecutable(
+  appResourcesDir: string,
+  appName: string,
+): NativefierOptions {
+  const options: NativefierOptions = {};
+  let executablePath: string | undefined = undefined;
+
+  const appRoot = path.resolve(path.join(appResourcesDir, '..', '..'));
+  const children = fs.readdirSync(appRoot, { withFileTypes: true });
+  const macOS =
+    children.filter((c) => c.name === 'MacOS' && c.isDirectory()).length > 0;
+  const windows =
+    children.filter((c) => c.name.toLowerCase().endsWith('.exe') && c.isFile())
+      .length > 0;
+  const linux =
```

Comment:
These three variables could have better names. Instead of `windows`, `macOS`, `linux`, what about `looksLikeWindows|macOS|linux`? (or something else you like better)